botnim/document_parser/dynamic_extractions/pdf_extraction/text_extraction.py

```diff
@@ -0,0 +1,165 @@
+import logging
+from pathlib import Path
+import argparse
+import sys
+from .exceptions import PDFTextExtractionError
+
+logger = logging.getLogger(__name__)
+
+def extract_text_with_pdfplumber(pdf_path: Path) -> str:
+    try:
+        import pdfplumber
+    except ImportError:
+        raise PDFTextExtractionError("pdfplumber is not installed. Please install it with 'pip install pdfplumber'.")
```

Comment:
1. Import at the top
2. No need for this. If we need a library, it needs to be installed.

Reply:
(relevant to all source files)